adafruit_avrprog.py

```diff
@@ -345,7 +353,7 @@ def _flash_word(self, addr, low, high):
         self._transaction((0x40, addr >> 8, addr, low))
         self._transaction((0x48, addr >> 8, addr, high))
 
-    def _flash_page(self, page_buffer, page_addr, page_size):
+    def _flash_page(self, page_buffer: float, page_addr: int, page_size: int):
```

Comment:
Looks like `page_buffer` needs to support access by indices and that individual elements are `int`.  Looking at how it gets used in the internals, it looks like it expects a `bytearray`.  You can also add the `None` return type!